### ISSUE TITLE
Problem resolving during self-update on first run

### ISSUE THREAD

**Issue by ChalyFlavour** [guisua/jdownloader-vpn-docker-synology#1]
Hey there,

thanks for doing all the work to bring this guide online.
While reading the troubleshooting I noticed that this may took quite some time..

Anway, I still encounter an issue attached below. TL;DR this is a name resolution error.
1. my /config directory (jdownloader container) is writeable and has some directories and files in it.
2. I tried to point it out by running a ping on "google.de" to test the basic DNS function
  docker exec -it nordvpn ping google.de
  docker exec -it jdownloader ping google.de
3. I checked the /etc/resolv.conf for the jdownloader container. It looked good, however I tried to set those values and restarted it without luck
nameserver 1.1.1.1
nameserver 8.8.8.8

You have any idea why the Java self-update doesnt resolve using direct connection?

Chears,
Chaly

`SEVERE [ org.jdownloader.update.UpdateManager.log ] -> Exception thrown at org.appwork.updatesys.client.defaultimpl.http.HttpClientImpl.log(HttpClientImpl.java:955):
org.appwork.utils.net.BasicHTTP.BasicHTTPException: org.appwork.utils.net.BasicHTTP.ReadIOException: java.net.UnknownHostException: Could not resolve: -update.appwork.org<->update.appwork.org-
        at org.appwork.utils.net.BasicHTTP.BasicHTTP.download(BasicHTTP.java:329)
        at org.appwork.utils.net.BasicHTTP.BasicHTTP.download(BasicHTTP.java:170)
        at org.appwork.utils.net.BasicHTTP.BasicHTTP.download(BasicHTTP.java:151)
        at org.appwork.updatesys.client.defaultimpl.http.HttpClientImpl.internalGetWithProxy(HttpClientImpl.java:870)
        at org.appwork.updatesys.client.defaultimpl.http.HttpClientImpl.get(HttpClientImpl.java:448)
        at org.appwork.updatesys.client.UpdateClient.get(UpdateClient.java:2255)
        at org.appwork.updatesys.client.UpdateClient.downloadPackage(UpdateClient.java:3987)
        at org.appwork.updatesys.client.UpdateClient.runUpdateCheck(UpdateClient.java:3980)
        at org.appwork.updatesys.client.UpdateClient.runPackageRequestAndServerSelection(UpdateClient.java:4090)
        at org.appwork.updatesys.client.UpdateClient.runUpdateCheck(UpdateClient.java:4054)
        at org.jdownloader.update.UpdateManager.runUpdateLoop(UpdateManager.java:1024)
        at org.jdownloader.update.PendingUpdate.run(PendingUpdate.java:19)
        at org.jdownloader.update.UpdateManager$23.run(UpdateManager.java:1677)
        at java.lang.Thread.run(Thread.java:748)
Caused by: org.appwork.utils.net.BasicHTTP.ReadIOException: java.net.UnknownHostException: Could not resolve: -update.appwork.org<->update.appwork.org-
`

**Comment by guisua** [comment]
Hi, I'm sorry but I can't tell right away what might be the issue. However, I've had problems with bubuntux/nordvpn in the past.

I will try to update the guide if I have time, but you may have more like using azinchen/nordvpn instead.

**Comment by ChalyFlavour** [comment]
> Hi, I'm sorry but I can't tell right away what might be the issue. However, I've had problems with bubuntux/nordvpn in the past.
> 
> I will try to update the guide if I have time, but you may have more like using azinchen/nordvpn instead.

Hey guisua
thank you for your reply. I’ve found the issue yesterday (using bubuntux) and did some monitoring  and fine tuning.

Primary this was a dns issue. If fixed it adding —dns using the IP of the bridge gateway of my Synology docker network (172.xx.0.1 in most cases).

After some hours my JD lost the internet connection again. I did some netcat inside the containers and noticed that my nordvpn container did a selfupdate and my JD stopped working. Fixed by restarting both containers. As I don’t want to monitor all that stuff, I’m fine with a planned task that restarts both containers every 8 hours. Yes, this is ugly workaround, but idc about 15seconds downtime. (I configured a planned task in the DSM7 GUI instead of using crontab for updating reasons.)
